src/indexer/services/substrateIndexer/substrateIndexer.ts

```diff
@@ -63,24 +137,107 @@ export class SubstrateIndexer {
     let latestBlock = await this.indexPastEvents()
     await this.provider.rpc.chain.subscribeNewHeads(async header => {
       // start at last block from past events query and move to new blocks range
-      if (latestBlock + this.currentEventsQueryInterval === Number(header.number)) {
+      if (latestBlock + this.currentEventsQueryInterval < Number(header.number)) {
         // connect executions to deposits
         try {
           // fetch and decode logs
+          const blockHash = await this.provider.rpc.chain.getBlockHash(latestBlock)
+          const signedBlock = await this.provider.rpc.chain.getBlock(blockHash)
+          const at = await this.provider.at(blockHash)
+          const timestamp = Number(await at.query.timestamp.now().toString())
+          const allRecords = await at.query.system.events()
+
+          // we get the proposal execution events - ts-ignore because of allRecords
+          // @ts-ignore-next-line
+          const proposalExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.ProposalExecution, allRecords) as Array<ProposalExecutionEvent>
+
+          // we get the deposit events - ts-ignore because of allRecords
+          // @ts-ignore-next-line
+          const depositEvents = getSubstrateEvents(SygmaPalleteEvents.Deposit, allRecords) as Array<DepositEvent>
+
+          // @ts-ignore-next-line
+          const failedHandlerExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.FailedHandlerExecution, allRecords) as Array<FailedHandlerExecutionEvent>
+
+          // we get the index of the section in the extrinsic
+          const sectionIndex = signedBlock.block.extrinsics.findIndex(ex => ex.method.section === "sygmaBridge")
+
+          // this is our identifier for the tx
+          const txIdentifier = `${latestBlock}-${sectionIndex}` //this is like the txHash but for the substrate
+
+          if (proposalExecutionEvents.length) {
```

Comment:
What's the difference between this logic and the logic above (meaning everything below `f (latestBlock + this.currentEventsQueryInterval < Number(header.number)) {`)? 
Seems it is the same so might as well extract into some functions.

Reply:
extracted